### PR TITLE
AC: Added processing different input folders for DIV2K dataset

### PR DESCRIPTION
**Pull request by Anna-Mironova**

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 1]
```diff
@@ -71,6 +77,8 @@ def parameters(cls):
 
     def configure(self):
         self.data_dir = self.get_value_from_config('data_dir')
+        self.lr_dir_suffix = self.get_value_from_config('lr_dir_suffix')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:80]: why is it named as suffix???

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 2]
```diff
@@ -43,6 +43,12 @@ def parameters(cls):
             'data_dir': PathField(
                 is_directory=True, description="Path to folder, where images in low and high resolution are located."
             ),
+            'lr_dir_suffix': StringField(
+                optional=True, description="Suffix of directory, where images in low resolution are located."
+            ),
+            'hr_dir_suffix': StringField(
+                optional=True, description="Suffix of directory, where images in high resolution are located."
+            ),
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:51]: it is not present in the original dataset but used in some cases, please provide also opportunity to set directory for upsample images

**Comment by eaidova** [comment]
@Anna-Mironova  please add new parameters to the readme

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 3]
```diff
@@ -84,7 +100,11 @@ def configure(self):
     def convert(self, check_content=False, progress_callback=None, progress_interval=100, **kwargs):
         content_errors = [] if check_content else None
         file_list_lr = []
-        for file_in_dir in self.data_dir.iterdir():
+        lr_dir = self.lr_dir if self.lr_dir else self.data_dir
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:103]: I think directories specification can be moved in configure() method.
and what happens if both provided?

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 4]
```diff
@@ -94,12 +114,17 @@ def convert(self, check_content=False, progress_callback=None, progress_interval
             lr_file_name = lr_file.parts[-1]
             upsampled_file_name = self.upsample_suffix.join(lr_file_name.split(self.lr_suffix))
             if self.two_streams and self.generate_upsample:
-                self.generate_upsample_file(lr_file, self.upsample_factor, upsampled_file_name)
+                dest_path = self.upsampled_dir if self.upsampled_dir else None
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:117]: probably upsample_dir can be configurated once, just use lr_dir as default

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 5]
```diff
@@ -94,12 +114,17 @@ def convert(self, check_content=False, progress_callback=None, progress_interval
...
+                if not hr_dir:
+                    content_errors.append('No one of the data_dir or hr_dir parameters are provided')
+                if hr_dir and not check_file_existence(hr_dir / hr_file_name):
                     content_errors.append('{}: does not exist'.format(self.data_dir / hr_file_name))
-                if self.two_streams and not check_file_existence(self.data_dir / upsampled_file_name):
+                upsampled_dir = self.upsampled_dir if self.upsampled_dir else lr_dir
+                if self.two_streams and not check_file_existence(upsampled_dir / upsampled_file_name):
                     content_errors.append('{}: does not exist'.format(self.data_dir / upsampled_file_name))
 
             identifier = [lr_file_name, upsampled_file_name] if self.two_streams else lr_file_name
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:130]: here need to be more accurate. If upsample and lr are stored in different directories

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 6]
```diff
@@ -71,10 +84,31 @@ def parameters(cls):
 
     def configure(self):
         self.data_dir = self.get_value_from_config('data_dir')
+        self.lr_dir = self.get_value_from_config('lr_dir') if self.get_value_from_config('lr_dir') else self.data_dir
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:87]:
```suggestion
        self.lr_dir = self.get_value_from_config('lr_dir')  or self.data_dir
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 7]
```diff
@@ -71,10 +84,31 @@ def parameters(cls):
...
+        if not self.lr_dir:
+            raise ConfigError('One of the parameters: data_dir or lr_dir should be provided for conversion')
+        self.hr_dir = self.get_value_from_config('hr_dir') if self.get_value_from_config('hr_dir') else self.data_dir
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:90]:
```suggestion
        self.hr_dir = self.get_value_from_config('hr_dir') or self.data_dir
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 8]
```diff
@@ -71,10 +84,31 @@ def parameters(cls):
...
         self.lr_suffix = self.get_value_from_config('lr_suffix')
         self.hr_suffix = self.get_value_from_config('hr_suffix')
         self.upsample_suffix = self.get_value_from_config('upsample_suffix')
         self.two_streams = self.get_value_from_config('two_streams')
+        if self.two_streams:
+            if not self.data_dir:
+                if self.lr_dir == self.upsampled_dir:
+                    self.data_dir = self.lr_dir
+                else:
+                    raise ConfigError('data_dir parameter should be provided for conversion as common part of paths '
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:103]: why here can not be specified relative to lr_dir?

File: tools/accuracy_checker/accuracy_checker/representation/super_resolution_representation.py [hunk 9]
```diff
@@ -14,11 +14,23 @@
...
+import numpy as np
+
+from ..data_readers import BaseReader
 from .image_processing import ImageProcessingAnnotation, ImageProcessingPrediction
 
 
 class SuperResolutionAnnotation(ImageProcessingAnnotation):
-    pass
+    @property
+    def value(self):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/representation/super_resolution_representation.py:25]: please move logic to the base class

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py [hunk 10]
```diff
@@ -71,10 +84,31 @@ def parameters(cls):
 
     def configure(self):
         self.data_dir = self.get_value_from_config('data_dir')
+        self.lr_dir = self.get_value_from_config('lr_dir') if self.get_value_from_config('lr_dir') else self.data_dir
+        if not self.lr_dir:
+            raise ConfigError('One of the parameters: data_dir or lr_dir should be provided for conversion')
+        self.hr_dir = self.get_value_from_config('hr_dir') if self.get_value_from_config('hr_dir') else self.data_dir
+        self.upsampled_dir = self.get_value_from_config('upsampled_dir')
+        if not self.upsampled_dir:
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:92]: this dir used only if we have two_streams in input, so no need to fill it in other cases

File: tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py (hunk 8 above)
Comment (Anna-Mironova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:104]: there is case when path relative to lr_dir can not be found.
Example: lr_dir = common_path/LR, upsampled_dir = common_path/UP

Comment (eaidova) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/super_resolution_converter.py:104]: your example is not issue
upsample_dir = lr_dir/../UP

**Comment by eaidova** [comment]
Jenkins please retry a build

**Pull request merged**